### PR TITLE
[RISCV][CostModel] Recommit VPIntrinsics have same cost as their non-vp counterparts

### PR DESCRIPTION
**Pull request by michaelmaitland**
This was reverted in commit 0abaf3caee88ae74def2c7000aff8e61b24634bb (#67178).
    This version of the patch includes a fix which was caused by
    vp-reductions having an extra start value argument which the non-vp
    counterparts did not have.

    There was a bug that occured due to VPReduction having a start value
    argument that the non-vp counterpart did not have. Once this bug was
    fixed, I discovered that non-vp reductions get costed using
    Intrinsic::not_intrinsic. This should be fixed by a future patch, but
    for now the VP version of the intrinsic is costed in the same way.

### CODE REVIEW

File: llvm/test/Analysis/CostModel/RISCV/rvv-intrinsics.ll [hunk 1]
```diff
@@ -206,10 +206,588 @@ define void @vp_fshl() {
...
+; CHECK-NEXT:  Cost Model: Invalid cost for instruction: %28 = call double @llvm.reduce.fadd.nxv4f64(double undef, <vscale x 4 x double> undef)
+; CHECK-NEXT:  Cost Model: Found an estimated cost of 18 for instruction: %29 = call double @llvm.vp.reduce.fadd.nxv8f64(double undef, <vscale x 8 x double> undef, <vscale x 8 x i1> undef, i32 undef)
+; CHECK-NEXT:  Cost Model: Invalid cost for instruction: %30 = call double @llvm.reduce.fadd.nxv8f64(double undef, <vscale x 8 x double> undef)
+; CHECK-NEXT:  Cost Model: Found an estimated cost of 35 for instruction: %31 = call double @llvm.vp.reduce.fadd.nxv16f64(double undef, <vscale x 16 x double> undef, <vscale x 16 x i1> undef, i32 undef)
+; CHECK-NEXT:  Cost Model: Invalid cost for instruction: %32 = call double @llvm.reduce.fadd.nxv16f64(double undef, <vscale x 16 x double> undef)
+; CHECK-NEXT:  Cost Model: Found an estimated cost of 0 for instruction: ret void
+;
+  call float @llvm.vp.reduce.fadd.v2f32(float undef, <2 x float> undef, <2 x i1> undef, i32 undef)
+  call float @llvm.reduce.fadd.v2f32(float undef, <2 x float> undef)
```
Comment (topperc) [inline llvm/test/Analysis/CostModel/RISCV/rvv-intrinsics.ll:601]: Isn't the intrinsic called llvm.vector.reduce.fadd?

Comment (topperc) [inline llvm/test/Analysis/CostModel/RISCV/rvv-intrinsics.ll:601]: Isn't this intrinsic called vector.reduce.fadd

Comment (michaelmaitland) [reply llvm/test/Analysis/CostModel/RISCV/rvv-intrinsics.ll]: Updated.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**